packages/engine-server/src/seed/service.ts

```diff
@@ -99,8 +104,24 @@ export class SeedService {
     return { seed };
   }
 
-  async cloneSeed({ seed }: { seed: SeedConfig }) {
-    const spath = await SeedUtils.clone({ wsRoot: this.wsRoot, config: seed });
+  /**
+   *
+   * @param branch - optional branch to clone from
+   * @returns
+   */
+  async cloneSeed({ seed, branch }: { seed: SeedConfig; branch?: string }) {
+    const wsRoot = this.wsRoot;
+    const id = SeedUtils.getSeedId(seed);
+    const spath = SeedUtils.seed2Path({ wsRoot, id });
+    fs.ensureDirSync(path.dirname(spath));
+    const git = simpleGit({ baseDir: wsRoot });
+    const options: {
+      "--track"?: string;
+    } = {};
+    if (branch) {
+      options["--track"] = branch;
+    }
+    await git.clone(seed.repository.url, spath, options);
```

Comment:
Any possibilities of name collisions here? i.e. if folder w/ name already exists.

Reply:
we are going with the npm approach where the name has to be globally unique. a seed name is `{publisher}.{name}`